examples/chip-tool/commands/clusters/ModelCommand.cpp

```diff
@@ -85,6 +85,22 @@ void ModelCommand::ClearICDEntry(const chip::ScopedNodeId & nodeId)
     }
 }
 
+void ModelCommand::AddICDEntry(chip::app::ICDClientInfo & clientInfo, chip::ByteSpan key)
+{
+    CHIP_ERROR err = CHIPCommand::sICDClientStorage.SetKey(clientInfo, key);
+    if (err == CHIP_NO_ERROR)
+    {
+        err = CHIPCommand::sICDClientStorage.StoreEntry(clientInfo);
+    }
+
+    if (err != CHIP_NO_ERROR)
+    {
+        CHIPCommand::sICDClientStorage.RemoveKey(clientInfo);
+        ChipLogError(chipTool, "Failed to persist symmetric key with error: %" CHIP_ERROR_FORMAT, err.Format());
+        return;
+    }
```

Comment:
this logic can be simplified to only have 1 conditionnal block. either with an if else or with if(...return)...

Reply:
we need check the error from CHIPCommand::sICDClientStorage.StoreEntry, and we still need two blocks?